precompile/contracts/feemanager/contract_test.go

```diff
@@ -67,12 +77,27 @@ var (
 
 				return input
 			},
-			SuppliedGas: SetFeeConfigGasCost,
+			SuppliedGas: SetFeeConfigGasCost + FeeConfigChangedEventGasCost,
 			ReadOnly:    false,
 			ExpectedRes: []byte{},
 			AfterHook: func(t testing.TB, state contract.StateDB) {
 				feeConfig := GetStoredFeeConfig(state)
 				require.Equal(t, testFeeConfig, feeConfig)
+
+				logsTopics, logsData := state.GetLogData()
+				require.Len(t, logsTopics, 1)
+				require.Len(t, logsData, 1)
+
+				topics := logsTopics[0]
+				require.Len(t, topics, 2)
+				require.Equal(t, FeeManagerABI.Events["FeeConfigChanged"].ID, topics[0])
+				require.Equal(t, allowlist.TestEnabledAddr.Hash(), topics[1])
+
+				logData := logsData[0]
+				oldFeeConfig, resFeeConfig, err := UnpackFeeConfigChangedEventData(logData)
+				require.NoError(t, err)
+				require.True(t, zeroFeeConfig.Equal(&oldFeeConfig), "expected %v, got %v", zeroFeeConfig, oldFeeConfig)
+				require.True(t, testFeeConfig.Equal(&resFeeConfig), "expected %v, got %v", testFeeConfig, resFeeConfig)
```

Comment:
could we abstract this to a helper function instead of repeating it?

Reply:
good idea 👍